catalogs/catalog-hadoop/src/test/java/com/datastrato/gravitino/catalog/hadoop/integration/test/HDFSKerberosIT.java

```diff
@@ -0,0 +1,120 @@
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Tag("gravitino-docker-it")
+public class HDFSKerberosIT {
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSKerberosIT.class);
+
+  private static final ContainerSuite containerSuite = ContainerSuite.getInstance();
+  private static final String CLIENT_PRINCIPAL = "cli@HADOOPKRB";
+  private static UserGroupInformation clientUGI;
+
+  private static String keytabPath;
+
+  @BeforeAll
+  public static void setup() throws IOException {
+    containerSuite.startKerberosHiveContainer();
+
+    File baseDir = new File(System.getProperty("java.io.tmpdir"));
+    File file = Files.createTempDirectory(baseDir.toPath(), "test").toFile();
+    file.deleteOnExit();
+
+    // Copy the keytab and krb5.conf from the container
+    keytabPath = file.getAbsolutePath() + "/client.keytab";
+    containerSuite
+        .getKerberosHiveContainer()
+        .getContainer()
+        .copyFileFromContainer("/etc/admin.keytab", keytabPath);
+
+    String krb5TmpPath = file.getAbsolutePath() + "/krb5.conf_tmp";
+    String krb5Path = file.getAbsolutePath() + "/krb5.conf";
+    containerSuite
+        .getKerberosHiveContainer()
+        .getContainer()
+        .copyFileFromContainer("/etc/krb5.conf", krb5TmpPath);
+
+    // Modify the krb5.conf and change the kdc and admin_server to the container IP
+    String ip = containerSuite.getKerberosHiveContainer().getContainerIpAddress();
+    String content = FileUtils.readFileToString(new File(krb5TmpPath), StandardCharsets.UTF_8);
+    content = content.replace("kdc = localhost:88", "kdc = " + ip + ":88");
+    content = content.replace("admin_server = localhost", "admin_server = " + ip + ":749");
+    FileUtils.write(new File(krb5Path), content, StandardCharsets.UTF_8);
+
+    LOG.info("Kerberos kdc config:\n{}", content);
+
+    System.setProperty("java.security.krb5.conf", krb5Path);
+    System.setProperty("sun.security.krb5.debug", "true");
+  }
+
+  @AfterAll
+  public static void tearDown() {
+    // Reset the UGI
+    UserGroupInformation.reset();
+
+    // Clean up the kerberos configuration
+    System.clearProperty("java.security.krb5.conf");
+    System.clearProperty("sun.security.krb5.debug");
+  }
+
+  @Test
+  public void testKerberosHDFS() throws IOException {
+    Configuration conf = new Configuration();
+    conf.set("fs.defaultFS", defaultBaseLocation());
+    conf.setBoolean("fs.hdfs.impl.disable.cache", true);
+    conf.set("hadoop.security.authentication", "kerberos");
+
+    UserGroupInformation.setConfiguration(conf);
+    clientUGI = UserGroupInformation.loginUserFromKeytabAndReturnUGI(CLIENT_PRINCIPAL, keytabPath);
+    PrivilegedAction<?> action =
+        (PrivilegedAction)
+            () -> {
+              try {
+                FileSystem fs = FileSystem.get(conf);
+                Path path = new Path("/");
```

Comment:
You need to create a file and check the file owner to see if it is expected.

If kerberos is enabled, the user should be "cli" from my understanding.